usrtools/xskmap_load_send/xskmap_load_send.c

```diff
@@ -0,0 +1,226 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2022 Red Hat, Inc.
+ * Copyright (c) 2022 Intel Corporation.
+ */
+
+#include "xskmap_load_send.h"
+
+/* Long options start at 256 to distinguish from short options */
+#define OPT_NO_COLOR     "no-color"
+#define OPT_NO_COLOR_NUM 256
+#define HOST_NAME_LEN    32
+
+static int parse_args(int argc, char **argv);
+
+static void
+print_usage(char *prog_name)
+{
+    cne_printf("Usage: %s [-h] \n"
+               "  -L [level]     Enable a logging level\n"
+               "  -m <path>      The pinned xsk_map\n"
+               "  --%-12s Disable color output\n",
+               prog_name, OPT_NO_COLOR);
+}
+
+static int
+parse_args(int argc, char **argv)
+{
+    // clang-format off
+    struct option lgopts[] = {
+        {OPT_NO_COLOR, no_argument, NULL, OPT_NO_COLOR_NUM},
+        {NULL, 0, 0, 0}
+    };
+    // clang-format on
+    int opt, option_index;
+    char log_level[16] = {0};
+
+    /* Parse the input arguments. */
+    for (;;) {
+        opt = getopt_long(argc, argv, "hL:m:u:n:", lgopts, &option_index);
+        if (opt == EOF)
+            break;
+
+        switch (opt) {
+        case 'h':
+            print_usage(argv[0]);
+            return -1;
+        case 'L':
+            strlcpy(log_level, optarg, sizeof(log_level));
+            if (cne_log_set_level_str(log_level)) {
+                CNE_ERR("Invalid command option\n");
+                print_usage(argv[0]);
+                return -1;
+            }
+            break;
+        case 'm':
+            strlcpy(info.map_path, optarg, sizeof(info.map_path));
+            break;
+        case OPT_NO_COLOR_NUM:
+            tty_disable_color();
+            break;
+
+        default:
+            CNE_ERR("Invalid command option\n");
+            print_usage(argv[0]);
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+static int
+send_map_fd(int sock, int fd)
+{
+    char cmsgbuf[CMSG_SPACE(sizeof(int))] = {0};
+    struct msghdr msg                     = {0};
+    struct iovec iov;
+    char value[UDS_MAX_CMD_LEN] = {0};
+
+    snprintf(value, sizeof(value), "%s,%d", UDS_FD_ACK_MSG, fd);
+
+    iov.iov_base = &value;
+    iov.iov_len  = strnlen(value, sizeof(value));
+
+    msg.msg_name       = NULL;
+    msg.msg_namelen    = 0;
+    msg.msg_iov        = &iov;
+    msg.msg_iovlen     = 1;
+    msg.msg_flags      = 0;
+    msg.msg_control    = cmsgbuf;
+    msg.msg_controllen = CMSG_LEN(sizeof(int));
+
+    struct cmsghdr *cmsg = CMSG_FIRSTHDR(&msg);
+
+    cmsg->cmsg_level = SOL_SOCKET;
+    cmsg->cmsg_type  = SCM_RIGHTS;
+    cmsg->cmsg_len   = CMSG_LEN(sizeof(int));
+
+    *(int *)CMSG_DATA(cmsg) = fd;
+    int ret                 = sendmsg(sock, &msg, 0);
+
+    if (ret < 0)
+        CNE_ERR_RET_VAL(-errno, "Failed to send xsk_map fd with error=%s\n", strerror(errno));
+
+    return ret;
+}
+
+static int
+connect_host(uds_client_t *c, const char *cmd __cne_unused, const char *params __cne_unused)
```

Comment:
re checking uds_client_t *c -- I believe this is done through the client handler https://github.com/CloudNativeDataPlane/cndp/blob/main/lib/common/uds/uds.c#L475